File: near/src/main/java/com/adroitandroid/near/connect/server/TcpServerListener.java
package com.adroitandroid.near.connect.server;

import java.net.InetAddress;

public abstract class TcpServerListener {
    public void onStartFailure(Throwable e) {
        onServerStartFailed(e);
    }
    public abstract void onServerStartFailed(Throwable e);
    public abstract void onReceive(byte[] bytes, InetAddress inetAddress);
}

File: near/src/main/java/com/adroitandroid/near/connect/server/TcpServerListener.kt
package com.adroitandroid.near.connect.server

import java.net.InetAddress

abstract class TcpServerListener {
    fun onStartFailure(e: Throwable?) {
        onServerStartFailed(e)
    }
    abstract fun onServerStartFailed(e: Throwable?)
    abstract fun onReceive(bytes: ByteArray, inetAddress: InetAddress)
}